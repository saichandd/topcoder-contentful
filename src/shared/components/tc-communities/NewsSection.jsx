import PT from 'prop-types';
import React from 'react';
import Slider from 'react-slick';
import ArticleCard from './ArticleCard';
import Section from './Section';

export default function NewsSection({
  news,
  theme,
}) {
  if (!news || !news.length) return null;

  const settings = {
    dots: true,
    autoplay: false,
    infinite: false,
    lazyLoad: true,
    arrows: false,
    slidesToShow: 3,
    slidesToScroll: 1,
    className: theme.carouselContainer,
    dotsClass: theme.carouselDot,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          dots: true,
        },
      },
      {
        breakpoint: 768,
        settings: 'unslick',
      },
    ],
  };

  return (
    <Section
      theme={theme.section}
      title="Latest News"
    >
      <Slider {...settings}>
        {
          news.slice(0, 3).map((item, i) => (
            <div data-index={i} key={item.title} className={theme.carouselParent}>
              <ArticleCard
                theme={theme.card}
                imageSrc={`/community-app-assets/themes/common/NewsSection/news-0${1 + i}.jpg`}
                link={{
                  title: 'Read More',
                  url: item.link,
                }}
                text={item.description}
                title={item.title}
              />
            </div>
          ))
        }
      </Slider>
    </Section>
  );
}

NewsSection.defaultProps = {
  news: null,
  theme: {
    section: {},
    card: {},
    carouselContainer: '',
    carouselParent: '',
    carouselDot: '',
  },
};

NewsSection.propTypes = {
  news: PT.arrayOf(PT.shape()),
  theme: PT.shape({
    section: PT.shape(),
    card: PT.shape(),
    carouselContainer: PT.string,
    carouselParent: PT.string,
    carouselDot: PT.string,
  }),
};
