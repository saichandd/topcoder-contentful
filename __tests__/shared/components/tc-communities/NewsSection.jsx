import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import NewsSection from 'components/tc-communities/NewsSection';

const mockDatas = [
  {
    news: [{
      link: 'http',
      title: 'title',
      description: 'description',
    }],
  }, {

  },
];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  _.forEach(mockDatas, (data) => {
    renderer.render((
      <NewsSection {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
