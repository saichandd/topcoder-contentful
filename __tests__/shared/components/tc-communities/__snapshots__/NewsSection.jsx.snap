// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<ThemedSection
  anchor={null}
  composeAdhocTheme="deeply"
  composeContextTheme="softly"
  link={null}
  mapThemrProps={[Function]}
  subTitle=""
  theme={Object {}}
  themePriority="adhoc-context-default"
  title="Latest News"
>
  <Slider
    arrows={false}
    autoplay={false}
    className=""
    dots={true}
    dotsClass=""
    infinite={false}
    lazyLoad={true}
    responsive={
      Array [
        Object {
          "breakpoint": 1024,
          "settings": Object {
            "dots": true,
            "slidesToShow": 3,
          },
        },
        Object {
          "breakpoint": 768,
          "settings": "unslick",
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={3}
  >
    <div
      className=""
      data-index={0}
    >
      <ThemedArticleCard
        composeAdhocTheme="deeply"
        composeContextTheme="softly"
        imageSrc="/community-app-assets/themes/common/NewsSection/news-01.jpg"
        link={
          Object {
            "title": "Read More",
            "url": "http",
          }
        }
        mapThemrProps={[Function]}
        text="description"
        theme={Object {}}
        themePriority="adhoc-context-default"
        title="title"
      />
    </div>
  </Slider>
</ThemedSection>
`;

exports[`Matches shallow shapshot 2`] = `null`;
